File: 06-bstpractice-Java/BST.java

class Node {
    public int value;
    public Node left, right;
    public Node(int value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

public class BST {
    public Node root;
    
    public BST(int value) {
        this.root = new Node(value);
    }

    public void insert(int value) {
    	// Your code goes here
        insert(root,value);
    }

    private void insert(Node node, int value) {
    	// Your code goes here

        if(node==null )
        {
            root=new Node(value);          
            // return root;
        }
        if(value<root.value)
        {
            root.left=insert(root.left,value);
        }
        System.out.println("suk: "+node.value);
    }

    public boolean search(int value) {
    	// Your code goes here
    	// return false;
        return search(root,value);
    }

    private boolean search(Node current, int value) {
    	// Your code goes here



        System.out.println("ch: "+current.value);
    	return false;
    }

}

File: 06-bstpractice-Python/bst.py
class Node(object):
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

class BST(object):
    def __init__(self, root):
        self.root = Node(root)

    def insert(self, new_val):
        # Your code goes here
        if self.root==None:
            self.root=Node(new_val)
        else:
            rootVal=self.root
            upperVal=self.root

            while(rootVal!=None):
                upperVal=rootVal
                if new_val<rootVal.value:
                    rootVal=rootVal.left
                else:
                    rootVal=rootVal.right
            if new_val<upperVal.value:
                upperVal.left=Node(new_val)
            else:
                upperVal.right=Node(new_val)





    def printSelf(self):
        # Your code goes here
        print(self.root)
        
    def search(self, find_val):
        # Your code goes here
        while self.root!=None:
            if self.root.value==find_val:
                flag=True
                return flag
            elif self.root.value<find_val:
                self.root=self.root.right
            else:
                self.root=self.root.left
        return False

